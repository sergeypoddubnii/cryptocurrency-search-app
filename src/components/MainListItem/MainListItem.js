import React from 'react';
import { useDispatch } from 'react-redux';
import favCryptoActions from '../../redux/favCryptocurrency/favCryptoActions';
import styled from 'styled-components';
import { toast } from 'react-toastify';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faPlus } from '@fortawesome/free-solid-svg-icons';
import withWrapperListItem from '../../hoc/withWrapperListItem';
import toastSetting from '../../services/toastSettings';

const Button = styled.button`
  background: transparent;
  color: green;
  font-size: 14px;
  line-height: 18px;
  width: 18px;
  padding: 0;
  border: none;
  outline: none;
`;

const MainListItem = ({ name, symbol, price, id, isFavorite }) => {
  const dispatch = useDispatch();

  const addToFavoriteHandler = () => {
    if (isFavorite) {
      toast.warn(`${name} has already added to favorite`, toastSetting);
      return;
    }
    const crypto = { name, symbol, price, id };
    return dispatch(favCryptoActions.addToFavorite(crypto));
  };

  return (
    <Button
      onClick={addToFavoriteHandler}
      type="button"
      title="add to favorite list"
    >
      <FontAwesomeIcon icon={faPlus} />
    </Button>
  );
};

export default withWrapperListItem(MainListItem);
