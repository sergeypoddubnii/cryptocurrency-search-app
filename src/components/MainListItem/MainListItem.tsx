import React, { FC } from 'react';
import { useDispatch } from 'react-redux';
import favCryptoActions from '../../redux/favCryptocurrency/favCryptoActions';
import styled from 'styled-components';
import { toast } from 'react-toastify';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faPlus } from '@fortawesome/free-solid-svg-icons';
import withWrapperListItem from '../../hoc/withWrapperListItem';
import toastSetting from '../../services/toastSettings';

const Button = styled.button`
  background: transparent;
  color: green;
  font-size: 14px;
  line-height: 18px;
  width: 18px;
  padding: 0;
  border: none;
  outline: none;
`;
interface MainListItemTypes {
  name: string;
  symbol: any;
  price: number;
  id: string;
  isFavorite: boolean;
}

const MainListItem = ({
  name,
  symbol,
  price,
  id,
  isFavorite,
}: MainListItemTypes) => {
  const dispatch = useDispatch();

  const addToFavoriteHandler = () => {
    if (isFavorite) {
      toast.warn(`${name} has already added to favorite`, toastSetting);
      return;
    }
    const crypto = { name, symbol, price, id };
    return dispatch(favCryptoActions.addToFavorite(crypto));
  };

  return (
    <Button
      onClick={addToFavoriteHandler}
      type="button"
      title="add to favorite list"
    >
      <FontAwesomeIcon icon={faPlus} />
    </Button>
  );
};

export default withWrapperListItem(MainListItem);
