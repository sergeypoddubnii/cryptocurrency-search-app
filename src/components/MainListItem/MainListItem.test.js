import React from 'react';
import { shallow } from 'enzyme';
import MainListItem from './MainListItem.tsx';

describe('MainListItem', () => {
	const favoriteListItem = shallow(<MainListItem />);
	it('should render properly', () => {
		expect(favoriteListItem).toMatchSnapshot();
	});
});
