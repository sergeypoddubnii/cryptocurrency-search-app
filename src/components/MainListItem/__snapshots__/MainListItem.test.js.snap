// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainListItem should render properly 1`] = `
<styled.li>
  <styled.h2>
    Â 
  </styled.h2>
  <div>
    <span>
      $
    </span>
    <MainListItem />
  </div>
</styled.li>
`;
